ibis/formats/__init__.py

```diff
@@ -0,0 +1,175 @@
+from abc import ABC, abstractmethod
+from typing import Generic, TypeVar
+
+from ibis.expr.datatypes import DataType
+from ibis.expr.schema import Schema
```

Comment:
Perhaps these should be under a `TYPE_CHECKING` guard to avoid the cost of imports only used for type-checking/annotation.